pkg/ingestor/parser/slsa/parser_slsa.go

```diff
@@ -36,144 +41,217 @@ import (
+	if err := s.parseSlsaPredicate(doc.Blob); err != nil {
+		return err
 	}
-	s.getSubject(statement)
-	s.getMaterials(statement)
-	err = s.getSLSA(statement)
-	if err != nil {
+	if err := s.getSubject(); err != nil {
+		return err
+	}
+	if err := s.getMaterials(); err != nil {
 		return err
 	}
-	s.getBuilder(statement)
+	if err := s.getSLSA(); err != nil {
+		return err
+	}
+	s.getBuilder()
 	return nil
 }
 
-func (s *slsaParser) getSubject(statement *in_toto.ProvenanceStatement) {
+func (s *slsaParser) getSubject() error {
 	// append artifact node for the subjects
-	for _, sub := range statement.Subject {
-		artifacts := []*model.ArtifactInputSpec{}
-		for alg, ds := range sub.Digest {
-			artifacts = append(artifacts, &model.ArtifactInputSpec{
-				Algorithm: alg,
-				Digest:    strings.Trim(ds, "'"), // some slsa documents incorrectly add additional quotes
-			})
+	for _, sub := range s.header.Subject {
+		s.identifierStrings.UnclassifiedStrings = append(s.identifierStrings.UnclassifiedStrings, sub.Name)
+		se, err := getSlsaEntity(sub.Name, sub.Digest)
+		if err != nil {
+			return err
+		}
+		s.subjects = append(s.subjects, se)
+	}
+	return nil
+}
 
-			s.identifierStrings.UnclassifiedStrings = append(s.identifierStrings.UnclassifiedStrings, sub.Name)
+func (s *slsaParser) getMaterials() error {
+	switch s.header.PredicateType {
+	case slsa01.PredicateSLSAProvenance:
+		if err := s.getMaterials0(s.s01smt.Predicate.Materials); err != nil {
+			return err
+		}
+	case slsa02.PredicateSLSAProvenance:
+		if err := s.getMaterials0(s.s02smt.Predicate.Materials); err != nil {
+			return err
+		}
+	case slsa1.PredicateSLSAProvenance:
+		if err := s.getMaterials1(s.s1smt.Predicate.BuildDefinition.ResolvedDependencies); err != nil {
+			return err
 		}
-		s.subjects = append(s.subjects, getSlsaEntity(sub.Name, artifacts))
 	}
+	return nil
 }
 
-func (s *slsaParser) getMaterials(statement *in_toto.ProvenanceStatement) {
-	// append dependency nodes for the materials
-	for _, mat := range statement.Predicate.Materials {
-		artifacts := []*model.ArtifactInputSpec{}
-		for alg, ds := range mat.Digest {
-			artifacts = append(artifacts, &model.ArtifactInputSpec{
-				Algorithm: alg,
-				Digest:    strings.Trim(ds, "'"), // some slsa documents incorrectly add additional quotes
-			})
+func (s *slsaParser) getMaterials1(rds []slsa1.ResourceDescriptor) error {
+	for _, rd := range rds {
+		// Only one of URI, Digest, or Content is required.
+		// https://github.com/in-toto/attestation/blob/main/spec/v1.0/resource_descriptor.md
+		if len(rd.Digest) == 0 {
+			// Skip if Digest is empty
+			continue
+		}
+		if rd.URI == "" {
+			// If URI is empty, just add artifacts
+			s.bareMaterials = append(s.bareMaterials, getArtifacts(rd.Digest)...)
+			continue
+		}
+		// Digest(s) and URI are set, create IsOccurrence between them.
+		s.identifierStrings.UnclassifiedStrings = append(s.identifierStrings.UnclassifiedStrings, rd.URI)
```

Comment:
Is this behavior ok for ResolvedDependencies? see comments above and
https://github.com/in-toto/attestation/blob/main/spec/v1.0/resource_descriptor.md
https://slsa.dev/provenance/v1#builddefinition

Note: ResolvedDependencies contains old.materials and old.invocation.configSource